pkg/envoy/lds/inmesh.go

```diff
@@ -274,23 +276,20 @@ func (lb *listenerBuilder) getOutboundFilterChainMatchForService(dstSvc service.
 		return nil, err
 	}
 
-	contains := func(s []string, t string) bool {
-		for _, a := range s {
-			if a == t {
-				return true
-			}
-		}
-		return false
-	}
-
-	var endpointSet []string
+	endpointSet := mapset.NewSet()
 	for _, endp := range endpoints {
-		if !contains(endpointSet, endp.IP.String()) {
-			endpointSet = append(endpointSet, endp.IP.String())
-		}
+		endpointSet.Add(endp.IP.String())
 	}
 
-	for _, ip := range endpointSet {
+	// For deterministic ordering
+	sortedEndpoints := []string{}
+	endpointSet.Each(func(elem interface{}) bool {
+		sortedEndpoints = append(sortedEndpoints, elem.(string))
+		return false
+	})
+	sort.Strings(sortedEndpoints)
```

Comment:
Thank you @eduser25 
One day this could even become a function in a util package by itself - something like 'UniqueStringsFromSlice([]string) []string` or something.

Reply:
Actually, I tried with mapset but failed with ut, then change to linear search. Will look deeply next time. 😃